`test:verify-terminal` fails to build with `unix-2.8.*`

`test:verify-terminal` fails to build with `unix-2.8.1.0` (bundled with GHC 9.6.1):

```
$ cabal test verify-terminal -w ghc-9.6
Build profile: -w ghc-9.6.1 -O1
In order, the following will be built (use -v for more details):
 - vty-tests-5.37 (test:verify-terminal) (first run)
Preprocessing test suite 'verify-terminal' for vty-tests-5.37..
Building test suite 'verify-terminal' for vty-tests-5.37..
[1 of 1] Compiling VerifyOutput     ( programs/VerifyOutput.hs, /home/rscott/Documents/Hacking/Haskell/vty/dist-newstyle/build/x86_64-linux/ghc-9.6.1/vty-tests-5.37/t/verify-terminal/build/verify-terminal/VerifyOutput.o, /home/rscott/Documents/Hacking/Haskell/vty/dist-newstyle/build/x86_64-linux/ghc-9.6.1/vty-tests-5.37/t/verify-terminal/build/verify-terminal/VerifyOutput.dyn_o )

programs/VerifyOutput.hs:37:16: error: [GHC-83865]
    • Couldn't match expected type: OpenFileFlags
                                    -> IO System.Posix.Types.Fd
                  with actual type: IO System.Posix.Types.Fd
    • The function ‘openFd’ is applied to four value arguments,
        but its type ‘FilePath
                      -> OpenMode -> OpenFileFlags -> IO System.Posix.Types.Fd’
        has only three
      In a stmt of a 'do' block:
        nullOut <- openFd "/dev/null" WriteOnly Nothing defaultFileFlags
      In the expression:
        do nullOut <- openFd "/dev/null" WriteOnly Nothing defaultFileFlags
           t <- outputForConfig
                  $ defaultConfig
                      {outputFd = Just nullOut, termName = Just termName,
                       colorMode = Just NoColor}
           reserveDisplay t
           dc <- displayContext t (100, 100)
           ....
   |
37 |     nullOut <- openFd "/dev/null" WriteOnly Nothing defaultFileFlags
   |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

programs/VerifyOutput.hs:37:45: error: [GHC-83865]
    • Couldn't match expected type ‘OpenFileFlags’
                  with actual type ‘Maybe a0’
    • In the third argument of ‘openFd’, namely ‘Nothing’
      In a stmt of a 'do' block:
        nullOut <- openFd "/dev/null" WriteOnly Nothing defaultFileFlags
      In the expression:
        do nullOut <- openFd "/dev/null" WriteOnly Nothing defaultFileFlags
           t <- outputForConfig
                  $ defaultConfig
                      {outputFd = Just nullOut, termName = Just termName,
                       colorMode = Just NoColor}
           reserveDisplay t
           dc <- displayContext t (100, 100)
           ....
   |
37 |     nullOut <- openFd "/dev/null" WriteOnly Nothing defaultFileFlags
   |                                             ^^^^^^^
```

This is because the type of `unix`'s `openFd` function changed in `unix-2.8.0.0` to remove the `Maye FileMode` argument, instead making it a part of the `OpenFileFlags` argument. (See the [changelog](https://hackage.haskell.org/package/unix-2.8.0.0/changelog).)

I can see two different ways to fix this issue:

1. Guard the use of `Nothing` with `!MIN_VERSION_unix(2,8,0)`.
2. Replace the use of `unix`'s `openFd` with `base`' [`openFile`](https://hackage.haskell.org/package/base-4.18.0.0/docs/System-IO.html#v:openFile). As far as I can tell, nothing about this code requires anything that `openFile` couldn't give you.

Which would you prefer?

--- a/tests/programs/VerifyOutput.hs
+++ b/tests/programs/VerifyOutput.hs
@@ -1,6 +1,7 @@
 -- We setup the environment to invoke certain terminals of interest.
 -- This assumes appropriate definitions exist in the current environment
 -- for the terminals of interest.
+{-# LANGUAGE CPP #-}
 {-# LANGUAGE ScopedTypeVariables #-}
 module VerifyOutput where
 
@@ -34,7 +35,11 @@ smokeTestTermNonMac termName i = liftIOResult $ do
 
 smokeTestTerm :: String -> Image -> IO Result
 smokeTestTerm termName i = do
-    nullOut <- openFd "/dev/null" WriteOnly Nothing defaultFileFlags
+    nullOut <- openFd "/dev/null" WriteOnly
+#if !MIN_VERSION_unix(2,8,0)
+                      Nothing
+#endif
+                      defaultFileFlags
     t <- outputForConfig $ defaultConfig
         { outputFd = Just nullOut
         , termName = Just termName

